### PR TITLE
Add sdk_harness_log_level_overrides option for python sdk

### LINKED ISSUE #18222: Improve Logging Capabilities
[~tibor.kiss]
"""
Currently we are not instantiating the logger, just using the class what logging package provides.
Shortcoming of this approach is that the user cannot set the log level on a per module basis as all log messages
end up in the root level.
"""

Imported from Jira [BEAM-1825](https://issues.apache.org/jira/browse/BEAM-1825). Original Jira may contain additional context.
Reported by: altay.

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #18222

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/options/pipeline_options.py [hunk 1]
```diff
@@ -971,6 +971,20 @@ def _add_argparse_args(cls, parser):
...
+        '--sdk_harness_log_level_overrides',
+        action='append',
+        default=None,
+        help=(
+            'Controls the log levels for specifically named loggers. The '
+            'expected format is a json string: {\"module\":\"log_level\",...}. '
+            'For example, by specifying the value "{\"a.b.c\":\"DEBUG\"}, '
+            'the logger underneath the module "a.b.c" will be configured to '
+            'output logs at the DEBUG level. Similarly, by specifying the '
+            'value {"a.b.c":"WARNING"} all loggers underneath the "a.b.c" '
```
Comment (tvalentyn) [inline sdks/python/apache_beam/options/pipeline_options.py:984]: is escaping quotes necessary here as well?

Comment (Abacn) [reply sdks/python/apache_beam/options/pipeline_options.py:984]: yeah, not necessary. Copied from java sdk and found I even missed a quote mark. Will fix

File: sdks/python/apache_beam/runners/worker/sdk_worker_main.py [hunk 2]
```diff
@@ -268,6 +271,38 @@ def _get_log_level_from_options_dict(options_dict: dict) -> int:
...
+    return
+
+  for module_name, log_level in parsed_overrides.items():
+    try:
+      logging.getLogger(module_name).setLevel(log_level)
+    except Exception as e:
+      # Never crash the worker when exception occurs during log level setting
+      # but logging the error.
+      _LOGGER.error(
+          "Error occurs when setting log level for %s: %s", module_name, e)
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/worker/sdk_worker_main.py:303]:
```suggestion
          "Error occurred when setting log level for %s: %s", module_name, e)
```

File: sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py [hunk 3]
```diff
@@ -121,6 +124,52 @@ def test__get_log_level_from_options_dict(self):
...
+        (
+            # multiple overrides
+            ['{"a.b":"DEBUG"}', '{"c.d":"ERROR","c.d.e":15}'],
+            {"a.b": logging.DEBUG, "a.b.f": logging.DEBUG, "c.d": logging.ERROR,
+             "c.d.e": 15, "c.d.f": logging.ERROR}
+        )
+    ]
+    for case, expected in test_cases:
+      # Set log level overrides for this fake package name.
+      # This avoids pollutes real logging.
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py:145]:
```suggestion
      # This avoids polluting real logging.
```

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py:145]: Is it to just to avoid side-effects between `single` and `multiple` scenarios?

**Review by tvalentyn — APPROVED** [review]

**Comment by Abacn** [comment]
PTAL

**Pull request merged**